pkg/tecdsa/dkg/result.go

```diff
@@ -1,10 +1,55 @@
 package dkg
 
-import "github.com/keep-network/keep-core/pkg/tecdsa"
+import (
+	"fmt"
+
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"github.com/keep-network/keep-core/pkg/tecdsa"
+)
 
 // Result of distributed key generation protocol.
 type Result struct {
+	// Group represents the group state, including members, disqualified,
+	// and inactive members.
+	Group *group.Group
```

Comment:
I think we need it.
It is used [when we check is operator is a member of the group](https://github.com/keep-network/keep-core/blob/f29f201134e1db2d627c0e481cd2244cf4ea85a4/pkg/tecdsa/dkg/member.go#L442).

We could remove the `Misbehaved` from the `dkg.Result` structure, though.
We could filter the misbehaved (inactive or disqualified) members on the spot, when this data is needed.
It would of course be slower than caching it, but the code would be cleaner.

Reply:
You need it to be a part of the `signingMember` but I'm talking about the `Result` struct here. I think we don't need to return anything else than the `PrivateKeyShare` and `Misbehaved` array. Am I wrong?

Reply:
But the `signingMember` gets its `group` field from the `Result`.
But in theory we can remove `group` from the `Result` and return it from `n.dkgExecutor.Execute(...)` call.

Reply:
I think we did not understand each other. Let's break it down from the very beginning. DKG consists of two parts:
1. Actual protocol execution that produces the result
2. Result publication

What I mean is that part 1 should return the minimal required amount of information without redundancy. If we return the `Result` that holds key share and misbehaved + the `group` separately, we have redundancy as misbehaved can be inferred from the `group`. This is not something we should do here. If you need the `group` for part 2, we should do the following:

1. Part 1 should return `Result` that contains only `PrivateKeyShare` and `Group`. Nothing else. Misbehaved members can be taken from `Group`
2. Part 2 should take `Result` as parameter. It should use the `Group` to create the signing member. It should build the misbehaved array in the format expected by the on-chain contract.

Reply:
Removed the `Misbehaved` in d5d7ef1ed7ed6c7418216ab8728be35c3f5ad4c5.
Calculating the `Misbehaved` from the `Result.Group` could be done in a few ways. 
I opted for adding `MisbehavedMembersBytes()` function to the `Result`, so the misbehaved can be obtained anywhere they are needed. But it could very well be done differently.